homeassistant/components/sensor/vasttrafik.py

```diff
@@ -0,0 +1,118 @@
+
+REQUIREMENTS = ['vtjp==0.1.11']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_DELAY = 'delay'
+CONF_DEPARTURES = 'departures'
+CONF_FROM = 'from'
+CONF_HEADING = 'heading'
+CONF_KEY = 'key'
+CONF_NAME = 'name'
+CONF_SECRET = 'secret'
+
+ICON = 'mdi:train'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+PLANNER = None
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_KEY): cv.string,
+    vol.Required(CONF_SECRET): cv.string,
+    vol.Optional(CONF_DEPARTURES): [{
+        vol.Required(CONF_FROM): cv.string,
+        vol.Optional(CONF_DELAY, default=0): cv.positive_int,
+        vol.Optional(CONF_HEADING): cv.string,
+        vol.Optional(CONF_NAME): cv.string}]
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the departure sensor."""
+    from vasttrafik import JournyPlanner
+    global PLANNER
+    PLANNER = JournyPlanner(
+        config.get(CONF_KEY),
+        config.get(CONF_SECRET))
+    sensors = []
+    for departure in config.get(CONF_DEPARTURES):
+        sensors.append(
+            VasttrafikDepartureSensor(
+                departure.get(CONF_NAME),
+                departure.get(CONF_FROM),
+                departure.get(CONF_HEADING),
+                departure.get(CONF_DELAY)))
+    add_devices(sensors)
+
+
+class VasttrafikDepartureSensor(Entity):
+    """Implementation of a Vasttrafik Departure Sensor."""
+
+    def __init__(self, name, departure, heading, delay):
+        """Initialize the sensor."""
+        self._name = name or departure
+        self._departure = PLANNER.location_name(departure)[0]
+        self._heading = (PLANNER.location_name(heading)[0]
+                         if heading else None)
+        self._delay = timedelta(minutes=delay)
+        self._departureboard = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon for the frontend."""
+        return ICON
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if not self._departureboard:
+            return
+        departure = self._departureboard[0]
+        params = {
+            'Line': departure.get('sname', None),
+            'Track': departure.get('track', None),
+            'Direction': departure.get('direction', None),
+            'Accessibility': departure.get('accessibility', None)
+            }
+        return {k: v for k, v in params.items() if v}
+
+    @property
+    def state(self):
+        """Return the next departure time."""
+        if not self._departureboard:
+            _LOGGER.warning(
+                'No departures from "%s" heading "%s"',
+                self._departure['name'],
+                self._heading['name'] if self._heading else 'ANY')
+            return
+        return (self._departureboard[0]['rtTime']
```

Comment:
I am really not a big fan of these weird wrapped if statements. Can you unwrap it to make it more readable.